### ISSUE TITLE
Improve and clarify usage of navigable regions aria-label

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#7938]
The main section of the Gutenberg UI are wrapped in a few ARIA landmark regions, to help assistive technologies users to navigate the UI.

Each region needs to be labelled with an `aria-label` attribute. I'd like to propose to improve a bit the current labels and clarify their usage in the `components/higher-order/navigate-regions/README.md` where there's no mention of the labels.

Screenshot from the screen reader JAWS:

![screenshot 31](https://user-images.githubusercontent.com/1682452/42645443-4b9bff6e-85fe-11e8-898d-9c17b7cea099.png)

Note: the word "region" is added by JAWS.

1
To distinguish the Gutenberg ARIA landmarks from the ones used in the WordPress UI, the former ones  are prefixed with the word "Editor", except "Publish" which was recently added. I'd suggest to change it to "Editor Publish".

2
The "Editor toolbar" label is a bit confusing for two reasons:
- there are many toolbars in Gutenberg, it would be better to specify this is the "main" or the "top" toolbar (thinking at the option "Fix Toolbar to Top")
- nested within the "Editor toolbar" landmark, there's an "Editor toolbar" toolbar: using the same name for both is confusing

<img width="674" alt="screen shot 2018-07-12 at 18 26 56" src="https://user-images.githubusercontent.com/1682452/42646556-82dbe75c-8601-11e8-86dc-bf1cc6de90e9.png">

I'd suggest to rename the ARIA landmark region to "Editor top bar". Worth reminding these labels are not visible, they will be announced just by screen readers.

3
Improve the README.md explaining the usage of `aria-label`.
